### ISSUE TITLE
Rule for 'Sheik in Kakariko'

### ISSUE THREAD

**Issue by uncleawie** [AmazingAmpharos/OoT-Randomizer#147]
Hi AmazingAmpharos, 

Loving your work here, OOT has never been more fun.

I think there is a rules error with regard to 'Sheik in Kakariko'.

I am presently stuck on seed 47004156 (options: medallions, True-Random Colours, Carrot Refill, and create spoiler log -everything else default).  I have Forest, Fire and Water medallions from completing Deku, Dodongo and Jabu child dungeons,  but I cannot trigger Sheik in Kakariko as I have yet to be able to become an adult.  

[OoT_medallions_47004156_Spoiler.txt](https://github.com/AmazingAmpharos/OoT-Randomizer/files/2427585/OoT_medallions_47004156_Spoiler.txt)

The rule is:

```
set_rule(world.get_location('Sheik in Kakariko'),lambda state: state.has('Forest Medallion') and state.has('Fire Medallion') and state.has('Water Medallion'))
```

My monkey-see-monkey-do logic says I think it should read:

```
set_rule(world.get_location('Sheik in Kakariko'),lambda state: state.has('Forest Medallion') and state.has('Fire Medallion') and state.has('Water Medallion') and state.is_adult())
```

Of course when I make this change to the code above and use the same seed I get a completely different ROM so I have no idea if I fixed anything. 

I am kinda hoping I am wrong here, and that there is a way to trigger Sheilk's song in Kakariko. So please feel free to say this is not a bug. :-)

Cheers

Awie

**Comment by flagrama** [comment]
Yes, this is a known issue with 2.0 that will be fixed in 3.0 https://github.com/AmazingAmpharos/OoT-Randomizer/issues/54 https://github.com/AmazingAmpharos/OoT-Randomizer/issues/97

**Comment by uncleawie** [comment]
Oh cool. Thanks very much.
